webui/react/src/pages/ExperimentDetails/ExperimentTrials.settings.ts

```diff
@@ -44,7 +44,7 @@ export const DEFAULT_COLUMN_WIDTHS: Record<TrialColumnName, number> = {
 };
 
 export const isOfSortKey = (sortKey: React.Key): sortKey is V1GetExperimentTrialsRequestSortBy => {
-  return Object.values(V1GetExperimentTrialsRequestSortBy).includes(String(sortKey));
+  return (Object.values(V1GetExperimentTrialsRequestSortBy) as Array<string>).includes(String(sortKey));
```

Comment:
`Object.values(V1GetExperimentTrialsRequestSortBy).includes(sortKey as V1GetExperimentTrialsRequestSortBy)` is better

Reply:
`sortKey as V1GetExperimentTrialsRequestSortBy` is not really correct since `sortKey` is not guaranteed to be a `V1GetExperimentTrialsRequestSortBy` whereas `as Array<string>` is an upcast since `Array<string>` is a supertype of `Array<V1GetExperimentTrialsRequestSortBy>`.

Reply:
i dont think its completely wrong cuz if its wrong, TS complains like 
<img width="1202" alt="image" src="https://user-images.githubusercontent.com/109113616/205401885-b9741a68-461b-4082-b74f-c06219ace3e9.png">

also, object literal values might have other types other than string. in this case, `Array<string>` works tho. 

prob [this](https://www.typescriptlang.org/play?#code/MYewdgzgLgBAgjAvDA3gWAFA2zAhgLhgHIIATAMyIBpMcYAjQoiozAXzwhlEigG5MmKAE8ADgFN4SGCIkhy8ANoBrccPkyx4jXAC6AjJh7QZ4gB6xkzSjAi4oASwjkH4rnANHwEEABtxAHS+IADmABQA8vQAVuLAUAEAbri+AK5uYXAAlAEOYMBppBlQ5lBZWZhAA) would be the most accurate? 
```ts
const A = {
    a: 'sdf',
    b: 'df'
} as const;

type A = typeof A[keyof typeof A];

const text = 'df' satisfies A;

console.log(Object.values(A).includes(text))
```

Reply:
Typescript allows unsafe casts as long the types have some relationship, the point of casting in typescript is actually to subvert the type system which is why you want to avoid it.

You can see here how typescript will implicitly upcast types when given a type hint but will not downcast because downcasting is unsafe:
```typescript
type MyEnum = 'foo' | 'bar' | 'baz'

const foo: MyEnum = 'foo';
const foo2: string = foo;

const bar: string = 'bar'
const bar2: MyEnum = bar; // type error
```
The important thing to understand here is that as a structurally-typed language Typescript follows the Liskov substitution principle to relate types to each other. `MyEnum` is a subtype of `string` and because you can use a `MyEnum` anywhere you could use a string and thus it is always safe to cast from `MyEnum` to `string`.

The reverse is not true, there are many strings that would not work if used as a `MyEnum`.

Really the best way to write this code is:
```typescript
const values: Array<string> = Object.values(V1GetExperimentTrialsRequestSortBy);
return values.includes(sortKey);
```
which eliminates the cast.

Reply:
actually this is what we want, i guess
```ts
export const isOfSortKey = (sortKey: React.Key): sortKey is V1GetExperimentTrialsRequestSortBy => {
  return String(sortKey) in V1GetExperimentTrialsRequestSortBy;
};
```

i originally used Object.value cuz enum doesnt support `in` operator

Reply:
```ts
const values: Array<string> = Object.values(V1GetExperimentTrialsRequestSortBy);
return values.includes(sortKey);
```
this makes more sense.
i was tricked by my code. the prev code i wrote isnt readable

Reply:
Ended up with ```(Object.values(V1GetExperimentTrialsRequestSortBy) as Array<string>)```